### ISSUE TITLE
Constantly pinging index.php/204 but not auto uploading

### ISSUE THREAD

**Issue by Mazvy** [nextcloud/android#9598]
Clients have enabled auto upload on their devices, the apps are constantly pinging `index.php/204` in bursts but not auto uploading anything. 

Sometimes the clients do start to upload, but more often then not it requires them to open the app and only then do the files start to transfer over.

Seeing as this pinging is constant and periodic, I assume the background process is working and not put to sleep by Android, however auto uploads are more often than not - not getting through.

    xx.x.x.xxx - - [21/Dec/2021:17:51:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6640 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:51:17 +0200] "GET /index.php/204 HTTP/1.1" 204 6660 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:54:12 +0200] "GET /index.php/204 HTTP/1.1" 204 6676 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:54:12 +0200] "GET /index.php/204 HTTP/1.1" 204 6672 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:56:26 +0200] "GET /index.php/204 HTTP/1.1" 204 6654 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:56:26 +0200] "GET /index.php/204 HTTP/1.1" 204 6648 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:57:49 +0200] "GET /index.php/204 HTTP/1.1" 204 6676 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:17:57:50 +0200] "GET /index.php/204 HTTP/1.1" 204 6650 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6660 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6678 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6638 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6674 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6644 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xx.x.x.xxx - - [21/Dec/2021:18:00:24 +0200] "GET /index.php/204 HTTP/1.1" 204 6644 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:02:02 +0200] "GET /index.php/204 HTTP/1.1" 204 6652 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:15 +0200] "GET /index.php/204 HTTP/1.1" 204 6678 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:15 +0200] "GET /index.php/204 HTTP/1.1" 204 6652 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6670 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6672 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6676 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6652 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6640 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:16 +0200] "GET /index.php/204 HTTP/1.1" 204 6646 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"
    xxx.xxx.x.xxx - - [21/Dec/2021:18:05:17 +0200] "GET /index.php/204 HTTP/1.1" 204 6662 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.18.0"

**Comment by AlvaroBrey** [comment]
`/index.php/204` is the path used to [check for connectivity to the server](http://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/nextcloud/client/network/ConnectivityServiceImpl.java#L78-L78), for reference. In turn, that is called [every time a file is uploaded](http://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/owncloud/android/operations/UploadFileOperation.java#L740-L740). 

In summary, the logs are normal and don't mean anything. If autouploads are failing, more details are needed from the app side (logcat would be best).

**Comment by Mazvy** [comment]
> `/index.php/204` is the path used to [check for connectivity to the server](https://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/nextcloud/client/network/ConnectivityServiceImpl.java#L78-L78), for reference. In turn, that is called [every time a file is uploaded](https://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/owncloud/android/operations/UploadFileOperation.java#L740-L740).
> 
> In summary, the logs are normal and don't mean anything. If autouploads are failing, more details are needed from the app side (logcat would be best).

But more than 9 times out of 10 no files are getting uploaded - the clients just constantly spam `/index.php/204` but don't upload anything.

The users in question aren't very tech savvy, and it's been a struggle gettings app logs out of them. Personally I don't have this issue, but I'm seeing this `/index.php/204` spam with nothing uploaded from multiple users.

**Comment by AlvaroBrey** [comment]
> > `/index.php/204` is the path used to [check for connectivity to the server](https://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/nextcloud/client/network/ConnectivityServiceImpl.java#L78-L78), for reference. In turn, that is called [every time a file is uploaded](https://github.com/nextcloud/android/blob/9c7d212d52e34302386564ecc5435a7ed3255db4/src/main/java/com/owncloud/android/operations/UploadFileOperation.java#L740-L740).
> > In summary, the logs are normal and don't mean anything. If autouploads are failing, more details are needed from the app side (logcat would be best).
> 
> But more than 9 times out of 10 no files are getting uploaded - the clients just constantly spam `/index.php/204` but don't upload anything.
> 
> The users in question aren't very tech savvy, and it's been a struggle gettings app logs out of them. Personally I don't have this issue, but I'm seeing this `/index.php/204` spam with nothing uploaded from multiple users.

I insist, the `index.php/204` is normal. Each app will ping that route _every time_ it tries to upload a file. Your issue has nothing to do with that route. The problem is elsewhere (and I don't know where)

**Comment by Mazvy** [comment]
> I insist, the `index.php/204` is normal. Each app will ping that route _every time_ it tries to upload a file. Your issue has nothing to do with that route. The problem is elsewhere (and I don't know where), and can't know without logs or reproduction steps.

It seems we're not on the same page.

I'm not saying `/index.php/204` is an issue. What I'm saying is that clients are constantly pinging `/index.php/204` but not auto uploading. 

The reason I'm bringing up `/index.php/204` spam is that to me it appears as if the Nextcloud Android app service is working fine on their phones as evident by the constant `/index.php/204` pinging, however auto uploads are not coming through. In server logs it's a wall of `/index.php/204` bursts with no files uploaded. 

I'll try to get someone on a dev version of the app to get some logs. I really wish we had some logs in the regular Nextcloud app as this makes troubleshooting with less tech savvy users next to impossible.... I cannot reproduce this issue on my phone - all files are uploaded perfectly and my phone isn't bombarding the server with `/index.php/204` bursts.

**Comment by obel1x** [comment]
Please consider this to be changed. I have NC with Android <5 Clients and there is nothing to sync.

I have checked Serverlogs: index.php is pinged 116910 times each day! This is unneccessary at all.

**Comment by obel1x** [comment]
As this was driving me nuts, i reviewed an changed the code to not ping the server. Maybe someone can check this.

**Comment by obel1x** [comment]
There may be a new solution for this, check #10448

**Comment by AlvaroBrey** [comment]
Should be fixed by #10488, or at least greatly alleviated by it
